packages/contracts-bedrock/test/libraries/Encoding.t.sol

```diff
@@ -92,4 +92,68 @@ contract Encoding_Test is CommonTest {
 
         assertEq(txn, _txn);
     }
+
+    /// @dev Tests that encodeSetL1BlockValuesInterop succeeds.
+    function test_encodeSetL1BlockValuesInterop_succeeds(
+        uint32 _baseFeeScalar,
+        uint32 _blobBaseFeeScalar,
+        uint64 _sequenceNumber,
+        uint64 _timestamp,
+        uint64 _number,
+        uint256 _baseFee,
+        uint256 _blobBaseFee,
+        bytes32 _hash,
+        bytes32 _batcherHash,
+        uint256[] memory _dependencySet
+    )
+        external
+        pure
+    {
+        vm.assume(_dependencySet.length <= type(uint8).max);
+
+        Encoding.encodeSetL1BlockValuesInterop({
+            _baseFeeScalar: _baseFeeScalar,
+            _blobBaseFeeScalar: _blobBaseFeeScalar,
+            _sequenceNumber: _sequenceNumber,
+            _timestamp: _timestamp,
+            _number: _number,
+            _baseFee: _baseFee,
+            _blobBaseFee: _blobBaseFee,
+            _hash: _hash,
+            _batcherHash: _batcherHash,
+            _dependencySet: _dependencySet
+        });
```

Comment:
Ideally what this does is it uses `ffi` to pass the arguments to the go implementation and then it can be differentially tested against the solidity implementation. This would depend on getting the Go L1 block attribute PR merged first. See [this](https://github.com/ethereum-optimism/optimism/tree/develop/packages/contracts-bedrock/scripts/go-ffi) for details